### PR TITLE
Add how-to Guide for MSSQL operators

### LINKED ISSUE #11913: Add how-to Guide for MSSQL operators
**Description**

A guide that describes how to use all the operators for MSSQL (https://github.com/apache/airflow/tree/master/airflow/providers/microsoft/mssql) would be useful.

Other guides are available:
https://airflow.readthedocs.io/en/latest/howto/operator/index.html

Source code for those guides are at:
https://github.com/apache/airflow/tree/master/docs/howto/operator

Are you wondering how to start contributing to this project? Start by reading our [contributor guide](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst)

Best regards,
Kaxil

**Related Issues**

Missing guides for operators in Microsoft provider:
- Azure: https://github.com/apache/airflow/issues/11912
- WinRM: https://github.com/apache/airflow/issues/11914

### PR DESCRIPTION
**Pull request by Bowrna**
MSSQL Example DAG. I have appended the working code by testing in SQL server.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Cool. Looks good, I would like however to add one more thing (this might be a separate PR but it would be great to add it here).  

The comments `[START howto_operator_mssql]` and the like are used to generate HowTo documentation out of the examples. This is cool the example DAG works, but the biggest benefit of it is the documentaiton on using the operators extracts the snippets of that working example dags and puts them in "Howto" guides. For example see how it is done in MySQL: https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-mysql/operators.rst

It would be great to heave such HowTo document added. Can you add it please too ?

**Comment by Bowrna** [comment]
> Cool. Looks good, I would like however to add one more thing (this might be a separate PR but it would be great to add it here).
> 
> The comments `[START howto_operator_mssql]` and the like are used to generate HowTo documentation out of the examples. This is cool the example DAG works, but the biggest benefit of it is the documentaiton on using the operators extracts the snippets of that working example dags and puts them in "Howto" guides. For example see how it is done in MySQL: https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-mysql/operators.rst
> 
> It would be great to heave such HowTo document added. Can you add it please too ?

I will add it in this PR as it will make more sense. thank you @potiuk for your feedback

**Comment by Bowrna** [comment]
@potiuk I have added the docs and i tried to commit the doc into repository and the pre-commit build fails 

```Validate providers.yaml files............................................................Failed
- hook id: provider-yamls
- exit code: 1

Checking integration duplicates
Checking completeness of list of {sensors, hooks, operators}
 -- {sensors, hooks, operators} - Expected modules(Left): Current Modules(Right)
Checking for duplicates in list of {sensors, hooks, operators}
Checking completeness of list of transfers
 -- Expected transfers modules(Left): Current transfers Modules(Right)
Checking for duplicates in list of transfers
Checking connection classes belong to package
 -- Checking providers: present in code(left), mentioned in /Users/sathishkannan/code/airflow/.github/ISSUE_TEMPLATE/airflow_providers_bug_report.yml (right)
Checking doc files
 -- Checking document urls: expected(left), current(right)
    -- Items in the left set but not the right:
       '/docs/apache-airflow-providers-microsoft-mssql/operators.rst'
```
I could not understand this failure message. Can you help me to understand this?

**Comment by eladkal** [comment]
> @potiuk I have added the docs and i tried to commit the doc into repository and the pre-commit build fails
> 
> ```
> - hook id: provider-yamls
> - exit code: 1
> 
> Checking integration duplicates
> Checking completeness of list of {sensors, hooks, operators}
>  -- {sensors, hooks, operators} - Expected modules(Left): Current Modules(Right)
> Checking for duplicates in list of {sensors, hooks, operators}
> Checking completeness of list of transfers
>  -- Expected transfers modules(Left): Current transfers Modules(Right)
> Checking for duplicates in list of transfers
> Checking connection classes belong to package
>  -- Checking providers: present in code(left), mentioned in /Users/sathishkannan/code/airflow/.github/ISSUE_TEMPLATE/airflow_providers_bug_report.yml (right)
> Checking doc files
>  -- Checking document urls: expected(left), current(right)
>     -- Items in the left set but not the right:
>        '/docs/apache-airflow-providers-microsoft-mssql/operators.rst'
> ```
> 
> I could not understand this failure message. Can you help me to understand this?

Note that it fails on `hook id: provider-yamls` so this gives you context into where the problem is. Something that is related to the provider yaml. Possibly you also edited a howto guide (`.rst` file) but you did not add a corresponded entry in the provider `.yaml`

**Comment by Bowrna** [comment]
> > @potiuk I have added the docs and i tried to commit the doc into repository and the pre-commit build fails
> > ```
> > - hook id: provider-yamls
> > - exit code: 1
> > 
> > Checking integration duplicates
> > Checking completeness of list of {sensors, hooks, operators}
> >  -- {sensors, hooks, operators} - Expected modules(Left): Current Modules(Right)
> > Checking for duplicates in list of {sensors, hooks, operators}
> > Checking completeness of list of transfers
> >  -- Expected transfers modules(Left): Current transfers Modules(Right)
> > Checking for duplicates in list of transfers
> > Checking connection classes belong to package
> >  -- Checking providers: present in code(left), mentioned in /Users/sathishkannan/code/airflow/.github/ISSUE_TEMPLATE/airflow_providers_bug_report.yml (right)
> > Checking doc files
> >  -- Checking document urls: expected(left), current(right)
> >     -- Items in the left set but not the right:
> >        '/docs/apache-airflow-providers-microsoft-mssql/operators.rst'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I could not understand this failure message. Can you help me to understand this?
> 
> Note that it fails on `hook id: provider-yamls` so this gives you context into where the problem is. Something that is related to the provider yaml. Possibly you also edited a howto guide (`.rst` file) but you did not add a corresponded entry in the provider `.yaml`

@eladkal thank you. I could understand the context of the problem in provider yaml. I have edited the `.rst` file. But I am not sure what I have to add in the provider `.yaml` file. Do I have to increase the version or make any changes to `.yaml` file for adding `.rst` file? If there is any help/guide about this, it would be useful for me to get started.

**Comment by eladkal** [comment]
I can't say for sure because I can't see your code but you can check other providers for example Amazon:
https://github.com/apache/airflow/blob/main/airflow/providers/amazon/provider.yaml#L43
You need to set similar `how-to-guide:` entry.

**Comment by potiuk** [comment]
Yeah @Bowrna . The Doc errors are a but misleading and require a bit of deeper understanding :

You need to make sure that link to "example_dags" is added to the index - otherwise no-one will be able to find it :):

```
/docs/apache-airflow-providers-microsoft-mssql/index.rst
```

You can see `index.rst` in other providers that alredy have `example_dags` and do it similarly.

**Comment by Bowrna** [comment]
thank you @eladkal and @potiuk

File: docs/apache-airflow-providers-microsoft-mssql/connections/mssql.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    specific language governing permissions and limitations
+    under the License.
+
+
+
+.. _howto/connection:mssql:
+
+MSSQL Connection
+======================
+The MSSQL connection type enables connection to `MSSQL <https://www.microsoft.com/en-in/sql-server/>`__.
```
Comment (uranusjr) [inline docs/apache-airflow-providers-microsoft-mssql/connections/mssql.rst]: Maybe we should use the “official” name here, i.e. Microsoft SQL Server.

File: docs/apache-airflow-providers-microsoft-mssql/connections/mssql.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Default Connection IDs
+----------------------
+
+MSSQL Hook uses parameter ``mssql_conn_id`` for Connection IDs and the value of the parameter
+as ``mssql_default`` by default.
```
Comment (uranusjr) [inline docs/apache-airflow-providers-microsoft-mssql/connections/mssql.rst]:
```suggestion
MSSQL Hook uses parameter ``mssql_conn_id`` for the Connection ID. The default value
is ``mssql_default``.
```

**Comment by Bowrna** [comment]
I am not sure why there is a failure in CI checks. Do I have to fix something to get the check pass?

**Comment by eladkal** [comment]
@Bowrna doc build is failing. can you take a look?

**Comment by Bowrna** [comment]
> @Bowrna doc build is failing. can you take a look?

sure i will check

File: docs/apache-airflow-providers-microsoft-mssql/operators.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+.. exampleinclude:: /../../airflow/providers/microsoft/mssql/example_dags/example_mssql.py
+    :language: python
+    :start-after: [START mssql_operator_howto_guide_create_table_mssql_from_external_file]
+    :end-before: [END mssql_operator_howto_guide_create_table_mssql_from_external_file]
+
+
+Your ``dags/create_table.sql`` should look like this:
+
+::
```
Comment (uranusjr) [inline docs/apache-airflow-providers-microsoft-mssql/operators.rst]:
```suggestion
.. code-block::sql
```

for syntax highlighting

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**